### DISCUSSION TITLE
clang-tidy parameters through clangd? (To set line-filter for clang-tidy)

### DISCUSSION THREAD

**Discussion by daantimmer** [clangd/clangd#1402] [Q&A]
I've got clang-tidy integrated in my build pipeline through cmake. Clangd gives me the option to 'directly' integrate clang-tidy in to my "IDE" (Vscode). However, for our unit tests we use googletest, and by default (depening on enabled checks) clang-tidy will trip over the usage of the macro `TEST_F`.
Two workarounds exist (three really, but the third one is non-negotiatable):
- just add `// NOLINT` after the `TEST_F()` line
- add a clang-tidy command line parameter like: `--line-filter=[{"name":"gtest.h"}]`

I prefer the second one as it just simply ignores all warnings coming resulting from macros used in test code from gtest. But that only works for running clang-tidy from cmake, where I can add the parameters to clang-tidy.

But clangd does not seem to offer this solution? Do other people have suggestions/workaround/fixes or other information that I can use to streamline my devops a bit more in this regard? For now I'll just resort to the `// NOLINT` solution to atleast keep clang-tidy-through-clangd shut.

**Comment by i-ky** [comment]
Have you tried putting `--line-filter` (as `LineFilter`, perhaps) into `.clang-tidy` file?

**Reply by daantimmer** [reply]
Does not work:
> E[12:40:05.893] tidy-config error at /workspaces/[redacted]/.clang-tidy:21:0: unknown key 'LineFilter'
> E[12:40:05.893] Error parsing clang-tidy configuration in /workspaces/[redacted]/.clang-tidy: Invalid argument

**Reply by HighCommander4** [reply]
I would suggest filing a new issue at https://github.com/llvm/llvm-project/issues with the `clang-tidy` label asking to add support for specifying the `--line-filter` option via the `.clang-tidy` file, as these files are respected by clangd. I don't think forwarding command-line parameters from clangd to clang-tidy would be scalable.

**Reply by daantimmer** [reply]
Done, although I can't add my own labels :-)
